src/clients/Client.cc

```diff
@@ -187,7 +187,6 @@ Client::serverComplete()
     }
 
     completed = true;
-    originalRequest()->hier.stopPeerClock(true);
```

Comment:
We are removing this stopPeerClock() call so the only one left is in FwdState::stopAndDestroy(). Can we prove that without this call, serverComplete() will invoke it somewhere below? This is what should happen immediately without reforwarding and/or adaptation. With adaptation, I think we should rely on methods such as handleAdaptationCompleted() or handleAdaptationAborted(), which eventually call stopAndDestroy() either through normal cleanup via, e.g., completeForwarding() or abortion, e.g., by calling handleUnregisteredServerEnd(). However it is difficult to track every single such path - especially if, theoretically, there might be an unexpected scenarios when adaptation does not call its handleAdaptationCompleted() at all.
Can we just state that FwdState::stopAndDestroy() is the only method dropping the 'self' back to nil and hence is required to be paired with FwdState::start()? If stopAndDestroy() is not called for some reason - there are problems more serious than peer time logging.

Reply:
> Can we guarantee that stopAndDestroy() will be called at the right time?

For the analysis, I think we should rely on at least some prerequisites, e.g., that the 'communication end' results in Client::serverComplete() (i.e., that the method documentation is correct). We should omit the 'reforwarding' path here (the communication is not finished in this case). Without adaptation, as I noted above, stopAndDestroy() will be called immediately by means of Client::completeForwarding() and FwdState::complete(). The adaptation case is more tricky because there are lots of adaptation-related Client methods, e.g., Client::handleAdaptation*(). These methods have a similar structure:
```
Client::anAdaptationMethod()
{
    if (aProblem) {
        abort();  // e.g., abortOnBadEntry() or abortAll()
        return;
    }
    // e.g., rep->body_pipe != nullptr, entry->isAccepting(), adaptedBodySource->exhausted(), doneWithAdaptation()  conditions
    if (waiting())
        return;
    success(); // e.g., handleAdaptationCompleted() or completeForwarding()
}
```
success() path should call stopAndDestroy() immediately via completeForwarding().
abort() path either waits for more data or results in the Client job termination via abortAll(), mustStop() (subsequently calling stopAndDestroy()).
waiting() path means that there is more data coming and hence were are not at the 'communication end' state yet.

Reply:
Thank you for this analysis. It gives me enough information to conclude that

* stopAndDestroy() must be called at the right time. If stopAndDestroy() is not called at the right time, it is a bug.
* We have studied current code and could not find any such bugs, but we cannot check every code path due to code complexities. If such a bug exists, fixing it can be considered outside this PR scope.